### PR TITLE
WT-10064 Don't evict internal pages with non-globally visible deleted pages for non standalone build

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: src/reconcile/rec_child.c [hunk 1]
```diff
@@ -125,6 +133,33 @@ __rec_child_deleted(
...
+             * don't write the page_del information, we don't have sufficient information to restore
+             * the page's information if subsequently read (we wouldn't know which transactions
+             * should see the original page and which should see the deleted page).
+             */
+            if (F_ISSET(r, WT_REC_EVICT))
+                return (__wt_set_return(session, EBUSY));
+
+            /*
+             * It is wrong to leave the page clean in checkpoint if we cannot write the deleted
+             * pages to disk in eviction. If we do so, the next eviction will write the non-globally
```
Comment (kommiharibabu) [inline src/reconcile/rec_child.c:158]: `to disk in eviction` to `to disk in checkpoint`.

How about `the next eviction will lose the time point information of non-obsolete deleted pages`?

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Review by agorrod — COMMENTED** [review]
I'm clearly still not understanding something related to this change.
Fast deleted pages do have version information associated with them now - unless the flag is set to explicitly retain the old format.
Why do we need to restrict eviction of internal pages when version information is included with the reconciliation?

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Review by agorrod — APPROVED** [review]
LGTM - the lack of testing is a bit concerning. We have WT-9801 to enable more testing of this configuration in WiredTiger, and I believe we need to make that better before this fix makes it into a release.

**Comment by quchenhao** [comment]
evergreen merge

**Pull request merged**